db/channel_cache.go

```diff
@@ -101,10 +101,12 @@ func newChannelCache(dbName string, terminator chan bool, options ChannelCacheOp
 		activeChannels:       activeChannels,
 		cacheStats:           cacheStats,
 	}
-	err := NewBackgroundTask("CleanAgedItems", dbName, channelCache.cleanAgedItems, options.ChannelCacheAge, terminator)
+	done, err := NewBackgroundTask("CleanAgedItems", dbName, channelCache.cleanAgedItems, options.ChannelCacheAge,
```

Comment:
Is there a channel cache close where we can wait for this to close?

Reply:
There is no channel cache close where we can wait for this to close, but we can close it while closing the database context.